nwb_conversion_tools/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -18,46 +24,24 @@ class BaseSortingExtractorInterface(BaseDataInterface, ABC):
     @classmethod
     def get_source_schema(cls):
         """Compile input schema for the SortingExtractor."""
-        return get_schema_from_method_signature(cls.__init__)
+        return get_schema_from_method_signature(cls.SX.__init__)
 
     def __init__(self, **source_data):
         super().__init__(**source_data)
         self.sorting_extractor = self.SX(**source_data)
 
     def get_metadata_schema(self):
         """Compile metadata schema for the RecordingExtractor."""
-        metadata_schema = super().get_metadata_schema()
-
-        # Initiate Ecephys metadata
-        metadata_schema["properties"]["Ecephys"] = get_base_schema(tag="Ecephys")
-        metadata_schema["properties"]["Ecephys"]["required"] = []
-        metadata_schema["properties"]["Ecephys"]["properties"] = dict(
-            UnitProperties=dict(
-                type="array",
-                minItems=0,
-                renderForm=False,
-                items={"$ref": "#/properties/Ecephys/properties/definitions/UnitProperties"},
-            ),
-        )
-        # Schema definition for arrays
-        metadata_schema["properties"]["Ecephys"]["properties"]["definitions"] = dict(
-            UnitProperties=dict(
```

Comment:
`UnitProperties` need to be in the schema, otherwise many tests are failing

Reply:
I think this is a wrong use of defining unit properties: we are currently ([example here](https://github.com/catalystneuro/nwb-conversion-tools/blob/e64d6f7fdd96ac82d7f5803f3f5a9ed294c9bfa2/nwb_conversion_tools/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py#L31)) defining unit properties along with data in the metadata which should not be the case. Subsequently, the `run_conversion()`, its writing those properties (pulling that data) into the units table. This is a wrong way to add unit properties: these should be added using the `set_unit_properties()` or `set_property()` method of the spikeextractors/spikeinterface in the `__init__`. This would automatically be pulled in the `add_units()` method then. 

Lets fix these changes in the child classes of `BaseSortingExtractorInterface` in another PR.